src/main/ipc.js

```diff
@@ -16,122 +16,110 @@ const setupNotifications = require('./notifications')
 const logHandler = require('./developerTools/logHandler')
 
 function init (cwd, state) {
-  const ipc = ipcMain
   const main = windows.main
   const dc = new DeltaChat(cwd, state.saved)
 
-  dc.on('ready', function () {
+  dc.on('ready', () => {
     if (!state.logins.includes(dc.credentials.addr)) {
       state.logins.push(dc.credentials.addr)
     }
   })
 
-  ipc.once('ipcReady', function (e) {
+  dc.on('DC_EVENT_IMEX_FILE_WRITTEN', filename => {
+    main.send('DC_EVENT_IMEX_FILE_WRITTEN', filename)
+  })
+
+  dc.on('DC_EVENT_IMEX_PROGRESS', progress => {
+    main.send('DC_EVENT_IMEX_PROGRESS', progress)
+  })
+
+  dc.on('error', error => main.send('error', error))
+
+  dc.on('DC_EVENT_LOGIN_FAILED', () => main.send('error', 'Login failed!'))
+
+  ipcMain.once('ipcReady', e => {
     app.ipcReady = true
     app.emit('ipcReady')
   })
 
-  ipc.on('setAspectRatio', (e, ...args) => main.setAspectRatio(...args))
-  ipc.on('setBounds', (e, ...args) => main.setBounds(...args))
-  ipc.on('setProgress', (e, ...args) => main.setProgress(...args))
-  ipc.on('show', () => main.show())
-  ipc.on('setAllowNav', (e, ...args) => menu.setAllowNav(...args))
-  ipc.on('chooseLanguage', (e, locale) => {
+  ipcMain.on('setAspectRatio', (e, ...args) => main.setAspectRatio(...args))
```

Comment:
It's easier to grep for `ipcMain` to see which ipc event listeners we have in the main process